src/components/TodoDetails.vue

```diff
@@ -0,0 +1,200 @@
+<script setup lang="ts">
+import { onMounted, ref } from "vue";
+import { useRoute, useRouter } from "vue-router";
+import useVuelidate from "@vuelidate/core";
+import { required, helpers } from "@vuelidate/validators";
+import { BASE_URL, TODO_NAME_RULES, Todo } from "../main";
+import axios from "axios";
+import TheLoader from "./TheLoader.vue";
+
+type TodoForm = {
+  todoName: string;
+  isComplete: boolean;
+  editing: boolean;
+};
+
+const route = useRoute();
+const router = useRouter();
+
+const displayLoader = ref(true);
+const todo = ref<Todo | undefined>();
+const editingTodo = ref<TodoForm>({
+  todoName: "",
+  isComplete: false,
+  editing: false,
+});
```

Comment:
I think it might be better to have a separate boolean ref for `editing` here because you're sort of tangling up your state for the todo info itself with information that is more about the state of the page and less about the todo